### PR TITLE
fix(patch): [sc-9383] Notify connection state properly for remote client endpoints.

### PR DESCRIPTION
**Pull request by ser-0xff**
## Description

Notify connection state properly for remote client endpoints.

## How Has This Been Tested?

Unit tests.

## Minimal checklist:

- [X] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [X] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 1]
```diff
@@ -692,21 +677,19 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
 
-    private func actorsForChannelLocked(_ channel: Channel) -> [(EndpointIdentifier, AsyncStream<InvocationEnvelope>.Continuation, ManagedAtomic<UInt64>)] {
+    private func actorsForChannelLocked(_ channelID: UInt32) -> [(EndpointIdentifier, AsyncStream<InvocationEnvelope>.Continuation, ManagedAtomic<UInt64>)] {
         // lock supposed to be locked by the caller
         var actors = [(EndpointIdentifier, AsyncStream<InvocationEnvelope>.Continuation, ManagedAtomic<UInt64>)]()
         for (actorID, actorInfo) in self.actors {
-            switch actorInfo {
-            case let .clientForRemoteService(cfrs):
-                if cfrs.channel === channel {
+            if actorID.channelID == channelID {
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:684]: `guard` + unite cases here too?

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 2]
```diff
@@ -958,25 +956,25 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
-                let ptr = Self.ptr(for: channel)
-                if var channelInfo = self.channels[ptr] {
-                    channelInfo.pendingSyncCalls.insert(callID)
-                    self.channels[ptr] = channelInfo
-                    logger.trace("add sync call \(callID) for \(ptr)")
-                } else {
-                    logger.error("internal error: channel \(ptr) not registered")
-                }
+                channelInfo.pendingSyncCalls.insert(callID)
+                self.channels[channelID] = channelInfo
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:973]: BTW, it's possible to update a dictionary element "in-place".

**Review by dimlio — APPROVED** [review]

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 3]
```diff
@@ -274,57 +275,47 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
-                case let .clientForRemoteService(cfrs):
-                    if cfrs.channel === channel {
+                if actorID.channelID == channelID {
+                    switch actorInfo {
+                    case let .clientForRemoteService(cfrs):
                         streamContinuations.append(cfrs.continuation)
-                    }
-                case let .serviceForRemoteClient(sfrc):
-                    if sfrc.channel === channel {
+                    case let .serviceForRemoteClient(sfrc):
```
[the remark this answers is not in the document]
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift]: Fixed.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 1 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:684]: Fixed.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 2 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:972]: I know, but better to keep code simpler from my point of view.
Hope compiler will do it properly :)

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 4]
```diff
@@ -274,57 +275,47 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
...
 
         let (streamContinuations, endpointContinuations, pendingSyncCalls) = lock.withLock {
             var streamContinuations = [AsyncStream<InvocationEnvelope>.Continuation]()
             var endpointContinuations = [CheckedContinuation<Void, Error>]()
             var actors: [EndpointIdentifier] = []
             for (actorID, actorInfo) in self.actors {
-                switch actorInfo {
-                case let .clientForRemoteService(cfrs):
-                    if cfrs.channel === channel {
+                if actorID.channelID == channelID {
```
[the remark this answers is not in the document]
Comment (dimlio) [reply Sources/DistributedSystem/DistributedSystem.swift:295]: Could also use
```swift
for (actorID, actorInfo) in self.actors where actorID.channelID == channelID {
```

**Pull request merged**